File: 디크리즈컨쿼_2/josephus.c
#include <stdio.h>
#include <stdlib.h>
void jose(int*,int,int);
void modify(int*,int);
int left;
int main(void){
    int N,CF;
    printf("인원 수를 입력하세요 : ");
    scanf("%d",&N);
    int* circle=(int*)malloc(sizeof(int)*N);
    for(int i=0;i<N;i++){
        circle[i]=i+1;
    }
    left=N;
    printf("constant factor를 입력하세요 : ");
    scanf("%d",&CF);
    jose(circle,CF-1,CF);
    free(circle);
    return 0;
}
void jose(int* arr, int start, int cf){
    modify(arr,start);
    int next=start+cf-1;
    while(next>=left){
        next-=left;
    }
    if(left==1){
        printf("마지막 생존자 : %d 번째 사람",arr[next]);
        return;
    }
    jose(arr,next,cf);
}
void modify(int* arr, int target){
    for(int i=target;i<left-1;i++){
        arr[i]=arr[i+1];
    }
    left--;
}

File: 디크리즈컨쿼_2/josephus.py
N=int(input("인원 수를 입력하세요 : "))
CF=int(input("constant factor를 입력하세요 : "))
circle=[]
for i in range(N):
    circle.append(i+1)
left=N
def modify(arr,target):
    global left
    for i in range(target,left-1):
        arr[i]=arr[i+1]
    arr.pop()
    left-=1
def jose(arr,start,cf):
    global left
    modify(arr,start)
    next=start+cf-1
    while next>=left:
        next-=left
    if left==1:
        print("마지막 생존자 :",arr[next],"번째 사람")
        return
    jose(arr,next,cf)
jose(circle,CF-1,CF)